components/automate-ui/src/app/pages/user-details/user-details.component.ts

```diff
@@ -45,56 +43,41 @@ export class UserDetailsComponent implements OnInit, OnDestroy {
     private fb: FormBuilder
   ) {
     this.createForms(this.fb);
-    // TODO (tc) This needs to be refactored to resemble our other patterns
-    // for specific object pages.
-    // combineLatest depends on the user object existing already.
-    this.user = {
-      id: '',
-      name: '',
-      membership_id: ''
-    };
-
-    this.done$ = <Observable<boolean>>combineLatest([
-      store.select(allUsers),
-      store.select(userStatus)])
-      .pipe(
-        map(([state, status]: [User[], EntityStatus]) => {
-          const id = this.user.id;
-          return status === EntityStatus.loadingSuccess && !find({ id }, state);
-        }));
-
-    this.subscriptions = [
-      store.select(userFromRoute).pipe(
-        filter(identity))
-        .subscribe((state) => {
-          this.user = <User>state;
-        }),
-      // Note: if the user browses directly to /settings/users/USERNAME, the state
-      // will not contain any user data -- so we need to fetch it.
-      store.select(routeParams).pipe(
-        pluck('id'),
-        filter(identity))
-        .subscribe((id: string) => {
-          store.dispatch(new GetUser({id}));
-        }),
-
-      // if the user is gone, go back to list
-      // note: we filter(identity) here -- so this will effectively only
-      // trigger when done$ is `true`
-      this.done$.pipe(filter(identity)).subscribe(
-        () => this.router.navigate(['/settings', 'users'])),
-
-      store.select(updateStatus).subscribe(this.handleUpdateStatus.bind(this))
-      ];
     }
 
   ngOnInit(): void {
+    this.store.pipe(
+      select(routeParams),
+      pluck('id'),
+      filter(identity),
+      takeUntil(this.isDestroyed))
+      .subscribe((id: string) => {
+        this.store.dispatch(new GetUser({ id }));
+      });
+
+    combineLatest([
+      this.store.select(userStatus),
+      this.store.select(userFromRoute)
+    ]).pipe(
+      filter(([status, user]) => status === EntityStatus.loadingSuccess && !isNil(user)),
+      takeUntil(this.isDestroyed))
+      .subscribe(([_, user]) => {
+        this.user = { ...user };
+      });
+
     this.route.data.subscribe((data: { isNonAdmin: boolean }) => {
       this.isAdminView = !data.isNonAdmin;
       // Update the forms once this info has come in
       this.createForms(this.fb);
       this.resetForms();
     });
+
+    this.store.select(updateStatus).subscribe(this.handleUpdateStatus.bind(this));
```

Comment:
Don't even remember the reason for splitting the setup from the handler, but I am pretty sure that reason is no longer relevant.
It would be cleaner to use our more conventional pattern--and also add in the required `takeUntil`:
```
    this.store.select(updateStatus).pipe(
      takeUntil(this.isDestroyed),
      filter(status => status === EntityStatus.loadingSuccess))
      // same status is used for updating password or full name, so we just reset both
      .subscribe(() => this.resetForms());

```
Then you can delete `handleUpdateStatus` entirely.